File: backend/internal/model/post.go
package model

import "time"

// Post 跑友动态
type Post struct {
	ID        string    `gorm:"type:char(36);primaryKey" json:"id"`
	UserID    string    `gorm:"type:char(36);not null;index:idx_post_user" json:"user_id"`
	RunID     *string   `gorm:"type:char(36);index:idx_post_run" json:"run_id,omitempty"`
	RouteID   *string   `gorm:"type:char(36);index:idx_post_route" json:"route_id,omitempty"`
	Content   string    `gorm:"type:text" json:"content"`
	CreatedAt time.Time `gorm:"type:datetime(3)" json:"created_at"`

	// 审核和可见性字段（由管理后台控制）
	ReviewStatus int8       `gorm:"type:tinyint;not null;default:0;index:idx_post_review" json:"review_status"`
	IsHidden     int8       `gorm:"type:tinyint;not null;default:0;index:idx_post_hidden" json:"is_hidden"`
	DeletedAt    *time.Time `gorm:"type:datetime(3);index:idx_post_deleted" json:"deleted_at,omitempty"`

	User  User   `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Run   *Run   `gorm:"foreignKey:RunID" json:"run,omitempty"`
	Route *Route `gorm:"foreignKey:RouteID" json:"route,omitempty"`

	// 计算字段（不在数据库中存储）
	LikeCount    int64 `gorm:"-" json:"like_count"`
	CommentCount int64 `gorm:"-" json:"comment_count"`
}

func (Post) TableName() string { return "posts" }

// PostComment 动态评论
type PostComment struct {
	ID        string    `gorm:"type:char(36);primaryKey" json:"id"`
	PostID    string    `gorm:"type:char(36);not null;index:idx_comment_post" json:"post_id"`
	UserID    string    `gorm:"type:char(36);not null;index:idx_comment_user" json:"user_id"`
	Content   string    `gorm:"type:text;not null" json:"content"`
	CreatedAt time.Time `gorm:"type:datetime(3)" json:"created_at"`

	User User `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

func (PostComment) TableName() string { return "post_comments" }

// PostLike 动态点赞
type PostLike struct {
	PostID    string    `gorm:"type:char(36);primaryKey" json:"post_id"`
	UserID    string    `gorm:"type:char(36);primaryKey" json:"user_id"`
	CreatedAt time.Time `gorm:"type:datetime(3)" json:"created_at"`
}

func (PostLike) TableName() string { return "post_likes" }

File: stride_moor_app/lib/core/models/post.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'user.dart';
import 'run.dart';
import 'route.dart';

part 'post.freezed.dart';

/// 跑友动态模型
@freezed
class Post with _$Post {
  const factory Post({
    required String id,
    required String userId,
    /// 关联跑步记录ID
    String? runId,
    /// 关联路线ID
    String? routeId,
    /// 动态文字内容
    String? content,
    /// 创建时间
    required DateTime createdAt,
    /// 发布者信息（Preload）
    User? user,
    /// 关联跑步记录（Preload）
    Run? run,
    /// 关联路线（Preload）
    Route? route,
    /// 点赞数（聚合字段）
    @Default(0) int likeCount,
    /// 评论数（聚合字段）
    @Default(0) int commentCount,
    /// 当前用户是否已点赞
    @Default(false) bool isLiked,
  }) = _Post;

  factory Post.fromJson(Map<String, dynamic> json) {
    User? user;
    final userJson = json['user'];
    if (userJson is Map<String, dynamic>) {
      user = User.fromJson(userJson);
    }

    Run? run;
    final runJson = json['run'];
    if (runJson is Map<String, dynamic>) {
      run = Run.fromJson(runJson);
    }

    Route? route;
    final routeJson = json['route'];
    if (routeJson is Map<String, dynamic>) {
      route = Route.fromJson(routeJson);
    }

    return Post(
      id: json['id'] as String,
      userId: json['user_id'] as String,
      runId: json['run_id'] as String?,
      routeId: json['route_id'] as String?,
      content: json['content'] as String?,
      createdAt: json['created_at'] == null
          ? DateTime.now()
          : DateTime.parse(json['created_at'] as String),
      user: user,
      run: run,
      route: route,
      likeCount: (json['like_count'] as num?)?.toInt() ?? 0,
      commentCount: (json['comment_count'] as num?)?.toInt() ?? 0,
      isLiked: (json['is_liked'] as bool?) ?? false,
    );
  }
}

/// 动态评论模型
@freezed
class PostComment with _$PostComment {
  const factory PostComment({
    required String id,
    required String postId,
    required String userId,
    required String content,
    required DateTime createdAt,
    User? user,
  }) = _PostComment;

  factory PostComment.fromJson(Map<String, dynamic> json) {
    User? user;
    final userJson = json['user'];
    if (userJson is Map<String, dynamic>) {
      user = User.fromJson(userJson);
    }

    return PostComment(
      id: json['id'] as String,
      postId: json['post_id'] as String,
      userId: json['user_id'] as String,
      content: json['content'] as String,
      createdAt: json['created_at'] == null
          ? DateTime.now()
          : DateTime.parse(json['created_at'] as String),
      user: user,
    );
  }
}
